### PR TITLE
Ticket 4 get comments by article

### PR DESCRIPTION
**Pull request by Shemz-ux**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -120,3 +122,42 @@ describe("GET /api/articles", () => {
...
+        expect(typeof comment.votes).toBe('number')
+        expect(typeof comment.created_at).toBe('string')
+        expect(typeof comment.author).toBe('string')
+        expect(typeof comment.body).toBe('string')
+        expect(typeof comment.article_id).toBe('number')
+      })
+    })
+  })
+
+  test("400: Returns an error due to an invalid request being made to the server", () => {
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:146]: I would make this test description clearer -  that the article_id is invalid

File: __tests__/app.test.js [hunk 2]
```diff
@@ -120,3 +122,42 @@ describe("GET /api/articles", () => {
...
+      const { comments } = body
+      expect(comments.length).toBeGreaterThan(0)
+      expect(comments).toBeSorted({ descending: true, key: 'created_at' })
+      comments.forEach((comment)=>{
+        expect(typeof comment.comment_id).toBe('number')
+        expect(typeof comment.votes).toBe('number')
+        expect(typeof comment.created_at).toBe('string')
+        expect(typeof comment.author).toBe('string')
+        expect(typeof comment.body).toBe('string')
+        expect(typeof comment.article_id).toBe('number')
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:141]: You should check that article_id is 3 here, they all should be the same!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -120,3 +122,42 @@ describe("GET /api/articles", () => {
...
+  })
+
+  test("404: Returns an error when asked to find a comment for an article id that does not exist", () => {
+    return request(app)
+    .get("/api/articles/1000/comments")
+    .expect(404)
+    .then(({body})=>{
+      expect(body.msg).toBe('Not found')
+    })
+  })
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:162]: One more test you can add, what if the article_id exists but has no comments yet? We want a 200 and an empty array back because in the future there could be comments added. You will need to check that the article exists
